api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -43,9 +43,19 @@ public interface ContentFile<F> {
    */
   FileContent content();
 
-  /** Returns fully qualified path to the file, suitable for constructing a Hadoop Path. */
+  /**
+   * Returns fully qualified path to the file, suitable for constructing a Hadoop Path.
+   *
+   * @deprecated since 1.7.0, will be removed in 2.0.0; use {@link #location()} instead.
+   */
+  @Deprecated
   CharSequence path();
 
+  /** Return the fully qualified path to the file, suitable for constructing a Hadoop Path */
```

Comment:
I wonder whether we still want to keep the Hadoop `Path` reference here as it quite evolved over the years and we do not necessarily depend on Hadoop anymore.

Reply:
Good point, I was largely concerned if we didn't preserve this property of being suitable for a hadoop path at the API level, once `path` is deprecated, it may break the users that do rely on having this. But if we're saying that we don't really need to concern ourselves on maintaining this, we may not need to include it. cc @rdblue thoughts on this?

Reply:
Discussed with @rdblue @danielcweeks and here are some key points on why we shouldn't document anything related to hadoop:

1. Hadoop pathing doesn't work with storage systems like GCS
2. Locations are a super set of what can be represented in a hadoop path. Even today `path` API can return paths which simply cannot be represented in a hadoop Path.
3. The hadoop path normalization logic simply conflicts with absolute paths 

So updated to remove this part of the comment.